app/models/merchant.rb

```diff
@@ -1,6 +1,16 @@
 class Merchant < ApplicationRecord
   has_many :items
   has_many :invoices
+  has_many :invoice_items, through: :items
   has_many :transactions, through: :invoices
   has_many :customers, through: :invoices
+
+  def self.most_items_sold(quantity)
+    Merchant.
+      select("merchants.*, SUM(invoice_items.quantity) AS most_items").
+      joins(:invoice_items).
+      order("most_items DESC").
+      group("merchants.id").
+      first(quantity)
```

Comment:
@pwentz Since the `quantity` value here will be coming from `params` in the controller, it would help to prevent against SQL injection by changing `.first(quantity)` to `.first("?", quantity)`. 

More info: http://guides.rubyonrails.org/active_record_querying.html#array-conditions